internal/getproviders/package_location_local_archive.go

```diff
@@ -56,13 +58,26 @@ func (p PackageLocalArchive) InstallProviderPackage(_ context.Context, meta Pack
 
 	filename := meta.Location.String()
 
-	// NOTE: We're not checking whether there's already a directory at
-	// targetDir with some files in it. Packages are supposed to be immutable
-	// and therefore we'll just be overwriting all of the existing files with
-	// their same contents unless something unusual is happening. If something
-	// unusual _is_ happening then this will produce something that doesn't
-	// match the allowed hashes and so our caller should catch that after
-	// we return if so.
+	// NOTE: Packages are immutable, but we may want to skip overwriting the existing
+	// files in due to specific scenarios defined below.
+
+	if entries, err := os.ReadDir(targetDir); err == nil && len(entries) != 0 {
```

Comment:
If we do the thing I mentioned in an earlier comment that prevents `Lock` from creating what appears to be a valid empty package directory, would that avoid the need for this unexplained `len(entries) != 0` check, or is that here for some other reason?

Reply:
Fixed in 0e670c6ad9